libtenzir/src/transfer.cpp

```diff
@@ -75,6 +68,26 @@ auto transfer::prepare(http::request req) -> caf::error {
       return err;
     }
   }
+  // Set the body.
+  if (not req.body.empty()) {
+    if (req.method == "POST") {
+      // Setting body data via CURLOPT_[COPY]POSTFIELDS implicitly sets the
+      // Content-Type header to 'application/x-www-form-urlencoded' unless we
+      // provide a Content-Type header that overrides it.
+      // TODO: Figure out a way to avoid the extra copy here, but this requires
+      // moving the request in some fashion and tying its lifetime to the
+      // handle.
+      if (auto err = to_error(easy.set(CURLOPT_COPYPOSTFIELDS, req.body))) {
+        return err;
+      }
+    } else if (req.method == "PUT") {
+      if (auto err = set(easy, chunk::make(std::move(req.body)))) {
+        return err;
+      }
+    } else {
+      TENZIR_WARN("unexpected request body with HTTP {} method", req.method);
```

Comment:
Note to the reviewer: this will fire on HTTP PATCH. But it's so incredibly rare that I didn't go the extra mile to see whether PATCH would go in to POST or PUT branch.